platforms/documentation/docs/src/snippets/bestPractices/avoidProviderGetOutsideTaskAction/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,33 @@
+// tag::provider-get-task-setup[]
+abstract class MyTask : DefaultTask() {
+    @get:Input
+    abstract val myInput: Property<String>
+
+    @get:OutputFile
+    abstract val myOutput: RegularFileProperty
+
+    @TaskAction
+    fun doAction() {
+        val outputFile = myOutput.get().asFile
+        val outputText = myInput.get() // <1>
+        println(outputText)
+        outputFile.writeText(outputText)
+    }
+}
+
+val currentEnvironment: Provider<String> = providers.gradleProperty("currentEnvironment") // <2>
+// end::provider-get-task-setup[]
```

Comment:
🤔 I think this example should include the declaration of the property inside the snippet that is shown.  Otherwise it isn't immediately obvious that `currentEnvironment` is a `Provider`, which weakens the example.  This keeps the origin of the provider out of the task wiring, which makes that part clearer, but still shows what it is.

Reply:
This snippet is shown as part of the set up. Are you asking to move it to the do and don't part so it is closer to where it is used? That's where it was originally, and you asked me to move it out of there.

Reply:
Sorry, this is my mistake, I wasn't looking at the rendered docs and only saw this had moved above the `avoid-this` and `do-this` tags.  This looks good.